src/lua/api-gateway/validation/oauth2/userProfileValidator.lua

```diff
@@ -81,6 +81,29 @@ function _M:getCacheToken(token)
     end
 end
 
+function _M:getCacheTokenLookupKey()
+    local oauth_token = ngx.var.authtoken
+    local oauth_token_hash = hasher.hash(oauth_token)
+    return self:getCacheToken(oauth_token_hash)
+end
+
+function _M:getRedisCacheLookupProfileKey()
+    if self.PROFILE_VALIDATOR_CODE ~= nil and self.PROFILE_VALIDATOR_CODE ~= "" then
+        return "user_json:" .. self.PROFILE_VALIDATOR_CODE;
+    else
+        return "user_json";
```

Comment:
We may have the same problem in the future if there's another user profile validator that does not define a PROFILE_VALIDATOR_CODE. We should either force each profile validator to define a PROFILE_VALIDATOR_CODE or rethink the solution (see also comments from this [PR](https://git.corp.adobe.com/adobe-apis/api-gateway-mc/pull/115).)

Reply:
the default is used here to have backwards compatibility with the current storing method.

Is a good idea to enforce a PROFILE_VALIDATOR_CODE. Another possible solution is to merge data for all profile validators flavours. Both these ideas will discussed in our team meetings.